File: TransactionRegister/Model/Support/TXError.m
//
//  Error.m
//  TransactionRegister
//
//  Created by Eric Romrell on 9/4/16.
//  Copyright © 2016 Eric Romrell. All rights reserved.
//

#import "TXError.h"

@implementation TXError

+(TXError *)errorWithError:(NSError *)error readableMessage:(NSString *)readableMessage andDebugMessage:(NSString *)debugMessage {
	TXError *me = [[self alloc] init];
	me.error = error;
	me.readableMessage = readableMessage;
	me.debugMessage = debugMessage;
	return me;
}

-(NSString *)description {
	return [NSString stringWithFormat:@"Error: %@\nReadable Message: %@\nDebug Message: %@", self.error, self.readableMessage, self.debugMessage];
}

@end

File: TransactionRegister/Model/Support/TXError.swift
//
//  TXError.swift
//  TransactionRegister
//
//  Created by Eric Romrell on 1/20/17.
//  Copyright © 2017 Eric Romrell. All rights reserved.
//

import Foundation

class TXError : CustomStringConvertible, Error {
	var error : Error?
	var readableMessage : String
	var debugMessage : String?
	
	init(error:Error?, readableMessage:String, debugMessage:String?) {
		self.error = error
		self.readableMessage = readableMessage
		self.debugMessage = debugMessage
	}
	
	public var description: String {
		return "Error: \(self.error)\nReadable Message: \(self.readableMessage)\nDebug Message: \(self.debugMessage ?? "nil")";
	}
}
